src/client.ts

```diff
@@ -231,20 +235,22 @@ in an environmental variable named FAUNA_SECRET or pass it to the Client\
         Authorization: `Bearer ${this.#clientConfiguration.secret}`,
         // WIP - typecheck should be user configurable, but hard code for now
         "x-typecheck": "false",
-        // WIP - presently core will default to tagged; hardcode to simple for now
-        // until we get back to work on the JS driver.
-        "x-format": "simple",
-        // include per-request headers
       };
       this.#setHeaders(
         { ...this.clientConfiguration, ...queryRequest },
         headers
       );
 
+      const isTaggedFormat =
+        (this.#clientConfiguration.format ?? "tagged") === "tagged" ||
+        queryRequest.format === "tagged";
```

Comment:
why not just use the derived headers? This is fine though.

Reply:
This way it stays typed, but I don't have a strong argument outside of that. I like this better, although it is more verbose.

Reply:
all good. that makes sense to me.